gobblin-metrics/src/main/java/gobblin/metrics/influxdb/InfluxDBReporter.java

```diff
@@ -121,50 +125,133 @@ public static Builder builder(String name) {
     return new Builder(name);
   }
 
-  private Serie.Builder addGauge(String name, Gauge gauge, long timeStamp, Serie.Builder builder) {
-    return builder.values(timeStamp, name, gauge.getValue());
+  private void reportGauge(MetricContext context, String name, Gauge gauge, long timeStamp) {
+    this.influxDB.write(
```

Comment:
Even if they are async, too many HTTP calls is never a good idea. An application could easily be sending thousands of metrics every minute, all creating independent HTTP requests. Performance would suffer from HTTP latency alone, and if there are many services, influxdb servers could be flooded with requests.